lighthouse-core/gather/gatherers/seo/font-size.js

```diff
@@ -316,16 +254,93 @@ class FontSize extends Gatherer {
-    } = await FontSize.fetchNodesToAnalyze(passContext.driver);
+    // We need to find all TextNodes that do not have legible text. DOMSnapshot.captureSnapshot is the
+    // fastest way to get the computed styles of every Node. Bonus, it allows for whitelisting properties.
+    // Once a bad TextNode is identified, its parent Node is needed. DOMSnapshot.captureSnapshot doesn't
+    // give the entire Node object, so DOM.getFlattenedDocument is used. The only connection between a snapshot
+    // Node and an actual Protocol Node is backendId, so that is used to join the two data structures.
+    const snapshot = await passContext.driver.sendCommand('DOMSnapshot.captureSnapshot', {
+      computedStyles: ['font-size'],
+    });
+
+    // Makes the strings access code easier to read.
+    /** @param {number} index */
+    const lookup = (index) => snapshot.strings[index];
+
+    // Locate the document under analysis.
+    // TODO: this needs to use frameId
+    const doc = snapshot.documents.find(doc => lookup(doc.documentURL) === passContext.url);
+
+    // doc is a flattened property list describing all the Nodes in a document, with all string values
+    // deduped in a strings array.
+    // Implementation:
+    // https://cs.chromium.org/chromium/src/third_party/blink/renderer/core/inspector/inspector_dom_snapshot_agent.cc?sq=package:chromium&g=0&l=534
+
+    if (!doc || !doc.nodes.nodeType || !doc.nodes.nodeName || !doc.nodes.backendNodeId
+      || !doc.nodes.nodeValue || !doc.nodes.parentIndex) {
+      throw new Error('Unexpected response from DOMSnapshot.captureSnapshot.');
+    }
+
+    // Not all nodes have computed styles (ex: TextNodes), so doc.layout.* is smaller than doc.nodes.*
+    // doc.layout.nodeIndex maps the index into doc.nodes.* to an index into doc.layout.styles.
+    // nodeIndexToStyleIndex inverses that mapping.
+    /** @type {Map<number, number>} */
+    const nodeIndexToStyleIndex = new Map();
+    for (let i = 0; i < doc.layout.nodeIndex.length; i++) {
+      nodeIndexToStyleIndex.set(doc.layout.nodeIndex[i], i);
+    }
+
+    /** @type {Map<number, {fontSize: number, textLength: number}>} */
+    const backendIdsToPartialFontData = new Map();
+    for (let i = 0; i < doc.nodes.nodeType.length; i++) {
+      const nodeType = doc.nodes.nodeType[i];
+      const nodeValue = lookup(doc.nodes.nodeValue[i]);
+      if (!isNonEmptyTextNode({
+        nodeType,
+        nodeValue,
+        parentNodeName: lookup(doc.nodes.nodeName[doc.nodes.parentIndex[i]]),
+      })) continue;
+
+      const styleIndex = nodeIndexToStyleIndex.get(doc.nodes.parentIndex[i]);
+      if (!styleIndex) continue;
+      const parentStyles = doc.layout.styles[styleIndex];
+      const [fontSizeStringId] = parentStyles;
+      const fontSize = parseInt(lookup(fontSizeStringId), 10);
+      backendIdsToPartialFontData.set(doc.nodes.backendNodeId[i], {
+        fontSize,
+        // TODO: trimming this for a second time. maybe don't?
```

Comment:
is that because it's internally being trimmed?

Reply:
because the check before (non empty text node) trims it too.

Reply:
could just make that function call check for a text node, and have the "non empty" part come after. I'll do that. Was mostly trying to keep the previous function as it was.

Reply:
but `.trim` doesn't mutate and we're not reassigning `nodeValue` so it should still be needed right? I don't think you have to refactor anything :)

Reply:
ooooooh I see what you were saying now 👍